### PR TITLE
Patch: lock exception

### LINKED ISSUE #946: job.result sometimes returns an empty dict when using save_result
The following remote program sometimes returns an empty dictionary as the result

```
# program.py
from quantum_serverless import save_result

my_list = [1, 2, 3, 4]
save_result({"results": my_list})
```

<Job | b90b6513-4a93-451b-871f-39681b26cbb7> returned an empty dictionary

### PR DESCRIPTION
**Pull request by IceKhan13**
### Summary

Yet another patch for lock.

### Details and comments

Fixes #946

### CODE REVIEW

**Comment by psschwei** [comment]
close / reopen to pull in latest changes for k8s test

**Comment by IceKhan13** [comment]
hm ,notebooks cannot connect to server still 🤔

**Comment by psschwei** [comment]
Strange, it was working fine in https://github.com/Qiskit-Extensions/quantum-serverless/actions/runs/6279180485 ... let's rerun and see (maybe it was something temporary)

**Comment by psschwei** [comment]
ahh, should've looked further down... the gateway pod isn't ready:

```
--- default pods ---
NAME                                READY   STATUS             RESTARTS      AGE
gateway-59b44bcf6d-cdskc            0/1     CrashLoopBackOff   4 (19s ago)   3m49s
keycloak-0                          1/1     Running            0             3m49s
kuberay-operator-6c498d996f-ftgfb   1/1     Running            0             3m49s
postgresql-0                        1/1     Running            0             3m49s
prepuller-5fjrq                     1/1     Running            0             3m49s
scheduler-55dc447bdd-cdt5f          1/1     Running            0             3m49s
```

**Comment by psschwei** [comment]
the restarts drop the logs, so going to work on debugging the failure in #972 ... but I wouldn't consider that blocking for this PR (i.e. feel free to override the k8s test)

**Comment by akihikokuroda** [comment]
One question.  Is the 'version' field of the 'job' updated after successful `job.save()`?

**Comment by IceKhan13** [comment]
@akihikokuroda yes, it basically [auto_update](https://docs.djangoproject.com/en/2.2/ref/models/fields/#django.db.models.DateField.auto_now)

**Comment by akihikokuroda** [comment]
I see.  Did you remove some `job.save()` because they are redundant?

**Comment by IceKhan13** [comment]
yes, they were redundant as we return object back. So, we only need to save in one place.

File: gateway/api/management/commands/schedule_queued_jobs.py [hunk 1]
```diff
@@ -67,6 +70,38 @@ def handle(self, *args, **options):
...
+                        attempts -= 1
+
+                        try:
+                            job.save()
+                            # remove artifact after successful submission and save
+                            if os.path.exists(job.program.artifact.path):
+                                os.remove(job.program.artifact.path)
 
+                            succeed = True
+                        except RecordModifiedError:
```
Comment (akihikokuroda) [inline gateway/api/management/commands/schedule_queued_jobs.py:92]: Sorry for silly questions.  I'm still trying to understand.  Does this exception happen when the `update_job_statuses` update the `job'?

Comment (IceKhan13) [reply gateway/api/management/commands/schedule_queued_jobs.py:92]: yes, those exception might happen anywhere where we have `job.save()`

**Review by Tansito — APPROVED** [review]

File: gateway/api/management/commands/schedule_queued_jobs.py (hunk 1 above)
Comment (Tansito) [reply gateway/api/management/commands/schedule_queued_jobs.py:92]: I don't see it ideal but by now I understand the fix at least. But yeah, I would like to see a TODO to improve it or something.

**Pull request merged**